include/bls12-381/fp.hpp

```diff
@@ -74,10 +87,10 @@ class fp2
     fp c1;
 
     fp2();
-    fp2(const std::array<fp, 2>& e2);
+    explicit fp2(const std::array<fp, 2>& e2);
```

Comment:
Why explicit?

Reply:
We only do the checks and Montgomery transform in deserialization methods. We and to reduce the probability of fp being constructed from raw array. explicit will help this slightly.